File: benchmarks/suite5/bits.go
package main


func popcount64(x int64) int64 {
	var c int64
	for x != 0 {
		c += x & 1
		x >>= 1
	}
	return c
}

func main() {
	const n int64 = 25000000
	x := int64(1)
	var acc int64
	for i := int64(0); i < n; i++ {
		x = (x*31 + i) % 1000000007
		acc += popcount64(x)
	}
	suite5PrintI64(acc)
}

File: benchmarks/suite5/bits.rs
mod bench_runtime;
use bench_runtime::suite5_print_i64;

fn popcount64(x: i64) -> i64 {
    let mut v = x;
    let mut c = 0i64;
    while v != 0 {
        c += v & 1;
        v >>= 1;
    }
    c
}

fn main() {
    let n: i64 = 25_000_000;
    let mut x: i64 = 1;
    let mut acc: i64 = 0;
    for i in 0..n {
        x = (x * 31 + i) % 1_000_000_007;
        acc += popcount64(x);
    }
    suite5_print_i64(acc);
}
